src/pygrambank/cldf.py

```diff
@@ -1,83 +1,191 @@
 import collections
+from itertools import groupby
+import re
 
 import pyglottolog
-from clldutils.misc import lazyproperty
+from clldutils.misc import lazyproperty, slug
 from pycldf.sources import Source
 
 from pygrambank import bib
-from pygrambank import srctok
-
-
-def bibdata(sheet, values, e, lgks, unresolved):
-    def clean_key(key):
-        return key.replace(':', '_').replace("'", "")
-
-    for row in values:
-        if row.Source:
-            row.Source_comment = row.Source
-            refs, sources = collections.OrderedDict(), []
-            uc = sum(list(unresolved.values()))
-            res = srctok.source_to_refs(row.Source, sheet.glottocode, e, lgks, unresolved)
-            if sum(list(unresolved.values())) > uc:  # pragma: no cover
-                row.Source_comment += ' (source not confirmed)'
-            for key, pages in res[0]:
-                typ, fields = e[key]
-                ref = key = clean_key(key)
-                if ref not in refs:
-                    refs[ref] = set()
-                refs[ref] = refs[ref].union(pages or [])
-                sources.append(Source(typ, key, **fields))
-
-            row.Source = [
-                '{}{}'.format(r, '[{}]'.format(','.join(sorted(p))) if p else '')
-                for r, p in refs.items()]
-            for src in sources:
-                yield src
-
-
-class Bibs(dict):
-    def __init__(self, glottolog, api):
-        dict.__init__(self, glottolog.bib('hh'))
-        self.update(api.bib)
-
-    def iter_codes(self):
-        for key, (typ, fields) in self.items():
-            if 'lgcode' in fields:
-                for code in bib.lgcodestr(fields['lgcode']):
-                    yield key, code
-
-
-def languoid_id_map(glottolog, glottocodes):
-    id_map = {}
-
-    for glottocode in glottocodes:
-        languoid = glottolog.api.languoid(glottocode)
-        lang = None
-
-        # Determine the associated language-level languoid:
-        if languoid.level.name == 'dialect':  # pragma: no cover
-            for _, gc, _ in reversed(languoid.lineage):
-                lang = glottolog.api.languoid(gc)
-                if lang.level.name == 'language':
-                    break
-        else:
-            lang = languoid
-
-        potential_ids = [
-            languoid.id, languoid.hid, languoid.iso, lang.id, lang.hid, lang.iso]
-        id_map.update((id_, glottocode) for id_ in potential_ids if id_)
-
-    return id_map
 
 
-def refs(api, glottocode, bibs, lgks, sheet):
-    def source(key):
-        type_, fields = bibs[key]
-        return key, type_, fields.get('author', fields.get('editor', '-')), fields.get('year', '-')
+MANUAL_SOURCE_MATCHES = {
```

Comment:
It might be useful to review these. Ideally, they should be resolved just like the other problems in the sheets.